### PR TITLE
feat(Scrollbar): new component

### LINKED ISSUE #232: [Feat]: New Scrollbar component
**Description**: The `Scrollbar `component provides a customizable scroll interface, typically used to enhance scrolling functionality and maintain design consistency across the application. This component is ideal for use in areas with overflow content, such as data tables, modal windows, or long content sections.

**Requirements**:

1. **Component Implementation:**

- Implement the `Scrollbar `component in TypeScript.
- Utilize SCSS with BEM methodology for styling.
- Integrate design tokens from @geneui/tokens for consistent colors, dimensions, and spacing.
- Support different orientations (e.g., horizontal, vertical) and customization options (e.g., track and thumb styling).

2. **Unit Tests:**

- Write unit tests using Jest and React Testing Library.
- Ensure test coverage for all props, including rendering of different orientations and custom styles.
- Test for edge cases such as content overflow, missing dimensions, and unsupported configurations.
- 

3. **Storybook Stories:**

- Create a Storybook story for the Scrollbar component.
- Showcase different configurations, such as horizontal and vertical scrollbars, and custom styles for the track and thumb.
- Create controls for the props so users can interact with the Scrollbar component in Storybook.
- Document all props of the public interfaces.

4. **Figma Design:**

Follow the design outlined in the provided [Figma link.](https://www.figma.com/design/XmW63z2fXfHBC95eCM36ed/Gene-UI%3A-Components?node-id=846-28579&p=f&m=dev)

**Checklist**:

- [ ]  Component is implemented in TypeScript.
- [ ]  Styling follows SCSS using BEM methodology.
- [ ]  Design tokens from @geneui/tokens are used for colors, dimensions, and spacing.
- [ ]  Unit tests written and cover different states and edge cases.
- [ ]  Storybook stories created with examples of all props and configuration options.
- [ ]  Figma design matches the component implementation.

### PR DESCRIPTION
**Pull request by GaroGabrielyan**
### Chromatic Build Summary 
**Build URL:** https://www.chromatic.com/build?appId=66daee7859a843dd87bfb3d7&number=212 
**Storybook URL:** https://66daee7859a843dd87bfb3d7-egcuhursmk.chromatic.com/

### CODE REVIEW

File: src/components/atoms/Scrollbar/Scrollbar.tsx [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    };
+
+    const hideScrollbarHandler = () => {
+        debouncedCallback();
+    };
+
+    // autoScrollTo Top and Left
+    useEffect(() => {
+        const scrollRefCurrent = scrollbarsRef.current;
+        if (scrollRefCurrent?.contentElement?.parentElement && (autoScrollTopTo || autoScrollLeftTo)) {
```
Comment (hamikhambardzumyan) [inline src/components/atoms/Scrollbar/Scrollbar.tsx:107]: Here is the condition will not work if I will use the value `0` for the `autoScrollTopTo` or `autoScrollLeftTo` and try to change checking something like `!== null` or `undefined`

File: src/components/atoms/Scrollbar/Scrollbar.stories.tsx
Comment (hamikhambardzumyan) [inline src/components/atoms/Scrollbar/Scrollbar.stories.tsx:1]: Please add accessibility support. 

https://www.w3.org/TR/wai-aria-1.2/#scrollbar

**Review by hamikhambardzumyan — CHANGES_REQUESTED** [review]

**Review by hamikhambardzumyan — CHANGES_REQUESTED** [review]

**Review by hamikhambardzumyan — APPROVED** [review]
Looks good to me
